server/src/test/java/org/elasticsearch/indices/settings/UpdateSettingsIT.java

```diff
@@ -435,23 +446,31 @@ public void testOpenCloseUpdateSettings() throws Exception {
         assertThat(getSettingsResponse.getSetting("test", "index.final"), nullValue());
     }
 
-    public void testEngineGCDeletesSetting() throws InterruptedException {
+    public void testEngineGCDeletesSetting() {
         createIndex("test");
         client().prepareIndex("test", "type", "1").setSource("f", 1).get();
         DeleteResponse response = client().prepareDelete("test", "type", "1").get();
         long seqNo = response.getSeqNo();
         long primaryTerm = response.getPrimaryTerm();
         // delete is still in cache this should work
         client().prepareIndex("test", "type", "1").setSource("f", 2).setIfSeqNo(seqNo).setIfPrimaryTerm(primaryTerm).get();
-        client().admin().indices().prepareUpdateSettings("test").setSettings(Settings.builder().put("index.gc_deletes", 0)).get();
+        assertAcked(client().admin().indices().prepareUpdateSettings("test").setSettings(Settings.builder().put("index.gc_deletes", 0)));
 
         response = client().prepareDelete("test", "type", "1").get();
         seqNo = response.getSeqNo();
-        Thread.sleep(300); // wait for cache time to change TODO: this needs to be solved better. To be discussed.
+
+        // Make sure the time has advanced for InternalEngine#resolveDocVersion()
+        for (ThreadPool tPool : internalCluster().getInstances(ThreadPool.class)) {
+            long time1 = tPool.relativeTimeInMillis();
+            long time2 = tPool.relativeTimeInMillis();
+            while (time1 == time2) {
```

Comment:
I think I'd have written this loop as the following, since there's no need for it to be a tight loop:

    assertBusy(() -> assertThat(tPool.relativeTimeInMillis(), greaterThan(time1)));

It is, admittedly, somewhat of a question of taste.